pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -641,12 +647,25 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 //  2. Kill pod sandbox if necessary.
 //  3. Kill any containers that should not be running.
 //  4. Create sandbox if necessary.
-//  5. Create ephemeral containers.
-//  6. Create init containers.
-//  7. Create normal containers.
-func (m *kubeGenericRuntimeManager) SyncPod(pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, backOff *flowcontrol.Backoff) (result kubecontainer.PodSyncResult) {
+func (m *kubeGenericRuntimeManager) SyncPodSandbox(pod *v1.Pod, podStatus *kubecontainer.PodStatus, pullSecrets []v1.Secret, backOff *flowcontrol.Backoff) (result kubecontainer.PodSyncResult, podIPs []string, podSandboxID string) {
```

Comment:
We don't need to return podIPs here.  determinePodSandboxIPs is called as part of runtime.GetPodStatus in kubelet.go SyncPod.  We should be able to directly use podStatus.IPs in the function below.

Reply:
In SyncPodSandbox():
```
		if !kubecontainer.IsHostNetworkPod(pod) {
			// Overwrite the podIPs passed in the pod status, since we just started the pod sandbox.
			podIPs = m.determinePodSandboxIPs(pod.Namespace, pod.Name, podSandboxStatus)
```
determinePodSandboxIPs() is called (and podIPs returned).

In kubelet:
```
	result, podIPs, sandboxID := kl.containerRuntime.SyncPodSandbox(pod, podStatus)
```
podIPs is used to see if we should update PodStatus with podIP.
So I think we should keep the podIPs return value.

Reply:
Right.  You can remove the entire first section, as it is already done by `runtime.GetPodStatus`.